### PR TITLE
Document how lock 'guard' structures are created.

### PR DESCRIPTION
**Pull request by frewsxcv**

### CODE REVIEW

File: src/libstd/sync/mutex.rs [hunk 1]
```diff
@@ -134,6 +134,13 @@ unsafe impl<T: ?Sized + Send> Sync for Mutex<T> { }
 ///
 /// The data protected by the mutex can be access through this guard via its
 /// `Deref` and `DerefMut` implementations
```
Comment (GuillaumeGomez) [inline src/libstd/sync/mutex.rs]: Not PR-bound but shouldn't be a dot present a the end of this sentence?

File: src/libstd/sync/rwlock.rs [hunk 2]
```diff
@@ -77,6 +77,13 @@ unsafe impl<T: ?Sized + Send + Sync> Sync for RwLock<T> {}
 
 /// RAII structure used to release the shared read access of a lock when
 /// dropped.
+///
+/// This structure is created by the [`read()`]  and [`try_read()`] methods on
```
Comment (GuillaumeGomez) [inline src/libstd/sync/rwlock.rs]: Extra whitespace in "[`read()`]  and".

**Review by GuillaumeGomez — CHANGES_REQUESTED** [review]
Just a small nit and it's good.

**Comment by frewsxcv** [comment]
Comments have been addressed.

**Comment by GuillaumeGomez** [comment]
Now squash time! \o/ (just the last commit)

**Review by GuillaumeGomez — APPROVED** [review]

**Comment by GuillaumeGomez** [comment]
Thanks!

@bors: r+ rollup

**Pull request merged**